### ISSUE TITLE
用7B-chat进行finetune为啥占用显存能够到30G以上，还报错”Attempting to unscale FP16 gradients.“ 有了解的吗

### ISSUE THREAD

**Issue by DJinsis** [QwenLM/Qwen#306]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by EnHuiPug** [comment]
30G哪里够finetune

**Comment by DJinsis** [comment]
> 30G哪里够finetune
只用训练lora都不够嘛，需要多大显存啊，我用的fp16，然后就报错

**Comment by DJinsis** [comment]
> 30G哪里够finetune

只用训练lora都不够嘛，需要多大显存啊，我用的fp16，然后就报错

**Comment by nlp4whp** [comment]
> > 30G哪里够finetune
> 
> 只用训练lora都不够嘛，需要多大显存啊，我用的fp16，然后就报错

fp16全量微调的话，粗略算，参数+梯度就28G了，AdamW又是全精度，至少100G吧

**Comment by EnHuiPug** [comment]
> > 30G哪里够finetune
> 
> 只用训练lora都不够嘛，需要多大显存啊，我用的fp16，然后就报错

不够，lora也需要40G+

**Comment by jonnyhe** [comment]
这边用qlora也报这个错，Attempting to unscale FP16 gradients.

**Comment by felixstander** [comment]
同报错

**Comment by jinqiupeter** [comment]
同样的错误，升级了peft和torch都不好

**Comment by Lesliefans0-0** [comment]
同问！我也是这个报错

**Comment by jonnyhe** [comment]
嗯，去掉这个注释可以跑起来了，但loss一直是零。。。。。

**Comment by JustinLin610** [comment]
建议大家使用最新的finetune.py以及finetune脚本，单卡qlora的问题已经解决。简单解释，就是此前单卡qlora走的fp16混合精度训练是用的torch的amp，那个会报issue提到的错。不能直接把fp16改掉，因为autogptq那边已经写死了得用fp16，现在更新的版本单卡和多卡保持一致，走的是deepspeed的，就可以正常训练了。另外最新的readme汇报了单卡lora和qlora在不同长度输入的情况下的显存占用和训练速度

**Comment by zjmvwb1** [comment]
> > > 30G哪里够finetune
> > 
> > 
> > 只用训练lora都不够嘛，需要多大显存啊，我用的fp16，然后就报错
> 
> 不够，lora可能也需要40G+

lora不就是给Q，K，V和转embedding部分加几个小矩阵，怎么可能多一倍数据量。

**Comment by Mary2333** [comment]
嗨，我也遇到了同样的报错，我是通过把peft的版本降低为0.4.0解决的，你可以试一下，peft=0.7.1的时候会报错。
